scripts/linux/agent_linux/functions.bash

```diff
@@ -0,0 +1,758 @@
+                
+                message="Agent argument 'proxy' is '$proxy'"
+                send_log_to_logzio "$LOG_LEVEL_DEBUG" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+                write_log "$LOG_LEVEL_DEBUG" "$message"
+
+                write_task_post_run "PROXY='$proxy'"
+                ;;
+            *)
+                message="agent.bash ($EXIT_CODE): unrecognized flag"
+                send_log_to_logzio "$LOG_LEVEL_ERROR" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+                write_task_post_run "write_error \"$message\""
+                message="agent.bash ($EXIT_CODE): try running the agent with '--help' flag for more information"
+                send_log_to_logzio "$LOG_LEVEL_ERROR" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+                write_task_post_run "write_error \"$message\""
+                
+                return $EXIT_CODE
+                ;;
+        esac
+    done
+}
+
+# Checks validation of the arguments
+# Input:
+#   ---
+# Output:
+#   if debug flag was used AGENT_ID='Debug'
+function check_arguments_validation {
+    local func_name="${FUNCNAME[0]}"
+
+    local message='Checking validation ...'
+    send_log_to_logzio "$LOG_LEVEL_DEBUG" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+    write_log "$LOG_LEVEL_DEBUG" "$message"
+
+    if [[ ! -z "$AGENT_JSON_FILE" ]]; then
+        if [[ -f "$AGENT_JSON_FILE" ]]; then
+            write_task_post_run "AGENT_ID='Debug'"
+            return
+        fi
+
+        message="agent.bash ($EXIT_CODE): the json file '$AGENT_JSON_FILE' does not exist"
+        send_log_to_logzio "$LOG_LEVEL_ERROR" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+        write_task_post_run "write_error \"$message\""
+
+        return $EXIT_CODE
+    fi
+
+    local is_error=false
+
+    if [[ -z "$APP_URL" ]]; then
+        is_error=true
+        message="agent.bash ($EXIT_CODE): Logz.io app url must be specified"
+        send_log_to_logzio "$LOG_LEVEL_ERROR" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+        write_task_post_run "write_error \"$message\""
+    fi
+    if [[ -z "$AGENT_ID" ]]; then
+        is_error=true
+        message="agent.bash ($EXIT_CODE): agent id must be specified"
+        send_log_to_logzio "$LOG_LEVEL_ERROR" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+        write_task_post_run "write_error \"$message\""
+    fi
+
+    if ! $is_error; then
+        return
+    fi
+
+    message="agent.bash ($EXIT_CODE): try running the agent with '--help' flag for more information"
+    send_log_to_logzio "$LOG_LEVEL_ERROR" "$message" "$LOG_STEP_PRE_INIT" "$LOG_SCRIPT_AGENT" "$func_name" "$AGENT_ID"
+    write_task_post_run "write_error \"$message\""
+
+    return $EXIT_CODE
+}
+
+# Download binary
```

Comment:
Just to keep with current conventions - can you add the comments before the function, that describe what it does, input, output, like in the other functions?